flat_pack.scad

```diff
@@ -0,0 +1,153 @@
+//openscad flat-pack joint library
+
+/*
+##module: outside_cuts
+Create a set of finger-joint cuts that result in two larger cuts taken at the outside
+edge
+  ###parameters:
+    *length* (real)         length of edge
+    *finger* (real)         length of each individual finger
+    *material* (real)       thickness of material - sets cut depth
+    *center* (boolean)         center the set of fingers with respect to origin
+*/
+
+module outside_cuts(length=6, finger=1, material=1, center=false) {
+  // overage to ensure that all cuts are completed
+  overage = 0.0001;
+
+
+  //maximum possible divisions for this length
+  max_divisions = floor(length/finger);
+  //number of usable divisions that fall completely within the edge
+  //for this implentation the number of divisions must be odd
+  usable_divisions = max_divisions%2==0 ? max_divisions-3 : max_divisions-2;
+
+  // number of "female cuts"
+  num_cuts = floor(usable_divisions/2);
+
+  //length of cut at either end
+  end_cut_length = (length-usable_divisions*finger)/2;
+
+  padding = end_cut_length + finger;
+
+  //set position relative to origin
+  x_translation = center==false ? 0 : -(num_cuts*2+1)*finger/2-end_cut_length;
+  y_translation = center==false ? 0 : -material/2;
+
+  translate([x_translation, y_translation]) {
+
+    // make both endcuts here
+    for (j = [0, 1]) {
+      translate([(length-end_cut_length)*j, -overage/2])
+        square([end_cut_length, material+overage]);
+    }
+
+
+    //make all the "normal" finger cuts here
+    for (i = [0 : num_cuts-1]) {
```

Comment:
I had to wrap this `for` statement in:

```
if (num_cuts > 0) {
  for (i = [0 : num_cuts-1]) {
// ...
```

as OpenSCAD was giving me errors when num_cuts was 0.

Reply:
Same is true with the similar statement in the other module.

Reply:
The if() solution works, but does it make sense to catch that? Shouldn't we get a warning when the number of cuts is 0? 
In what scenarios is it OK to have 0 cuts when calling a module that makes cuts?

Reply:
After thinking more about this, you're right. OpenSCAD doesn't like iterating over ranges that are negative and when num_cuts is 0, the for loop goes from `[0 : -1]` and that's not correct.